modules/library/src/main/scala/zio/elasticsearch/query/FunctionScoreFunction.scala

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright 2022 LambdaWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.elasticsearch.query
+
+import zio.Chunk
+import zio.elasticsearch.query.DecayFunctionType._
+import zio.elasticsearch.script.Script
+import zio.json.ast.Json
+import zio.json.ast.Json.{Num, Obj, Str}
+
+sealed trait FunctionScoreFunction {
+  val filter: Option[ElasticQuery[_]]
+  def withFilter(filter: ElasticQuery[_]): FunctionScoreFunction
+  private[elasticsearch] def toJson: Json
+}
+
+object FunctionScoreFunction {
+  def expDecayFunction(field: String, origin: String, scale: String): DecayFunction =
+    DecayFunction(
+      field = field,
+      decayFunctionType = Exp,
+      origin = origin,
+      scale = scale,
+      decay = None,
+      filter = None,
+      multiValueMode = None,
+      offset = None,
+      weight = None
+    )
+
+  def fieldValueFactor(fieldName: String): FieldValueFactor =
+    FieldValueFactor(
+      fieldName = fieldName,
+      factor = None,
+      filter = None,
+      modifier = None,
+      missing = None,
+      weight = None
+    )
+
+  def gaussDecayFunction(field: String, origin: String, scale: String): DecayFunction =
+    DecayFunction(
+      field = field,
+      decayFunctionType = Gauss,
+      origin = origin,
+      scale = scale,
+      decay = None,
+      filter = None,
+      multiValueMode = None,
+      offset = None,
+      weight = None
+    )
+
+  def linearDecayFunction(field: String, origin: String, scale: String): DecayFunction =
+    DecayFunction(
+      field = field,
+      decayFunctionType = Linear,
+      origin = origin,
+      scale = scale,
+      decay = None,
+      filter = None,
+      multiValueMode = None,
+      offset = None,
+      weight = None
+    )
+
+  def randomScoreFunction(): RandomScoreFunction =
```

Comment:
For consistency in code I prefer it to be function as it uses brackets when creating instance just like other methods.

```
functionScore(weightFunction(10.0)).withFunction(randomScore()).(fieldValueFactor("field"))
```
instead of
```
functionScore(weightFunction(10.0)).withFunction(randomScore).(fieldValueFactor("field"))
```

Reply:
@arnoldlacko